assembler/parse.c

```diff
@@ -159,11 +191,22 @@ int is_end_of_line(char c)
 {
 	return (c == '\n' || c == '\0' || c == EOF);
 }
-
-int is_keyword(char* str, int index)
+/*function that checks if the label or the command name is legal, that depends on the flag info.*/
+int is_keyword(char* str, int index, int flag)
 {
+	int max =0;
 	int i = 0;
-	for (i = 0; i < MAX_KEYWORDS; i++)
+	if (flag == CHECK_COMMAND_NAME)
```

Comment:
Hi, You can write a short line.
max =( flag == CHECK_COMMAND_NAME ? MAX_COMMAND_NAME  : MAX_KEYWORDS) ;